docs/sql-reserved-and-non-reserved-key-words.md

```diff
@@ -0,0 +1,573 @@
+---
+layout: global
+title: SQL Reserved/Non-Reserved Key Words
+displayTitle: SQL Reserved/Non-Reserved Key Words
+---
+
+When `spark.sql.parser.ansi.enabled` is set to true (false by default), some keywords are reserved for Spark SQL.
+The reserved keywords cannot be used for identifiers (e.g., table names, view names, column names, column aliases) in a query.
```

Comment:
do you know what's the definition of reserved keyword in SQL standard?

I checked the antlr file, seems the only difference between reserved and non-reserved keywords is, reserved keywords can't be used in table alias.

Note that, I treat all the keywords that can be used in `strictIdentifier` as reserved keywords.

Reply:
Yeah, I couldn't find the clear definition about what "reserved" means in any document I checked...

Reply:
anyway let's just document the current behavior in Spark: reserved keywords can't be used as table alias, and that's it.

Reply:
If we want to change the behavior of reserved keywords later, that's another story.

Reply:
Yes, it looks fine to me for now.